anvil/core/src/eth/serde_helpers.rs

```diff
@@ -115,6 +115,15 @@ where
     Ok(num)
 }
 
+pub fn lenient_block_number_array<'de, D>(deserializer: D) -> Result<Vec<BlockNumber>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let num =
+        <[Vec<LenientBlockNumber>; 1]>::deserialize(deserializer)?.into_iter().next().unwrap().into_iter().map(Into::into).collect();
```

Comment:
shouldn't be necessary to deserialize this as array of vec

instead `Vec::<Len..>::deserialize` should be sufficient

Reply:
Would the API then look like `eth_getBlocksByArray(1, 2, 3, 4, 5)` vs `eth_getBlocksByArray([1, 2, 3, 4, 5])`?

Reply:
oh good point, unsure about the ergonomics here,

either 

`params: [[1,2,3,4], true|false]`

or

`params: [1,2,3,4, true|false]`

I think the former would make more sense.

wdyt @sambacha

Reply:
I think `params: [[1,2,3,4], true|false]` is not correct based off looking at other similar methods.

e.g. here is how Besu request for `debug_storageRangeAt` works 



```json
{
    "jsonrpc": "2.0",
    "method": "debug_storageRangeAt",
    "params": [
        "0x2b76b3a2fc44c0e21ea183d06c846353279a7acf12abcc6fb9d5e8fb14ae2f8c",
        0,
        "0x0e0d2c8f7794e82164f11798276a188147fbd415",
        "0x0000000000000000000000000000000000000000000000000000000000000000",
        1
    ],
    "id": 1
}
```

Reply:
https://github.com/sambacha/safe-result/blob/master/src/error-handler/index.ts#L108-#L126